values_test.go

```diff
@@ -59,154 +60,196 @@ func newValueTestCases() map[string]valueTestCase {
 		},
 		"Int8": {
 			value:        NewInt8(-8),
 			string:       "-8",
+			goValue:      int8(-8),
 			expectedType: Int8Type{},
 		},
 		"Int16": {
 			value:        NewInt16(-16),
 			string:       "-16",
+			goValue:      int16(-16),
 			expectedType: Int16Type{},
 		},
 		"Int32": {
 			value:        NewInt32(-32),
 			string:       "-32",
+			goValue:      int32(-32),
 			expectedType: Int32Type{},
 		},
 		"Int64": {
 			value:        NewInt64(-64),
 			string:       "-64",
+			goValue:      int64(-64),
 			expectedType: Int64Type{},
 		},
 		"Int128": {
 			value:        NewInt128(-128),
 			string:       "-128",
+			goValue:      big.NewInt(-128),
 			expectedType: Int128Type{},
 		},
 		"Int256": {
 			value:        NewInt256(-256),
 			string:       "-256",
+			goValue:      big.NewInt(-256),
 			expectedType: Int256Type{},
 		},
 		"Word8": {
 			value:        NewWord8(8),
 			string:       "8",
+			goValue:      uint8(8),
 			expectedType: Word8Type{},
 		},
 		"Word16": {
 			value:        NewWord16(16),
 			string:       "16",
+			goValue:      uint16(16),
 			expectedType: Word16Type{},
 		},
 		"Word32": {
 			value:        NewWord32(32),
 			string:       "32",
+			goValue:      uint32(32),
 			expectedType: Word32Type{},
 		},
 		"Word64": {
 			value:        NewWord64(64),
 			string:       "64",
+			goValue:      uint64(64),
 			expectedType: Word64Type{},
 		},
 		"Word128": {
 			value:        NewWord128(128),
 			string:       "128",
+			goValue:      big.NewInt(128),
 			expectedType: Word128Type{},
 		},
 		"Word256": {
 			value:        NewWord256(256),
 			string:       "256",
+			goValue:      big.NewInt(256),
 			expectedType: Word256Type{},
 		},
 		"UFix64": {
 			value:        ufix64,
 			string:       "64.01000000",
+			goValue:      64.01000000,
 			expectedType: UFix64Type{},
 		},
 		"Fix64": {
 			value:        fix64,
 			string:       "-32.11000000",
+			goValue:      -32.11000000,
 			expectedType: Fix64Type{},
 		},
 		"Void": {
 			value:        NewVoid(),
 			string:       "()",
+			goValue:      nil,
 			expectedType: VoidType{},
 		},
 		"Bool": {
 			value:        NewBool(true),
 			string:       "true",
+			goValue:      true,
 			expectedType: BoolType{},
 		},
 		"some": {
 			value:        NewOptional(ufix64),
 			string:       "64.01000000",
+			goValue:      64.01000000,
 			expectedType: NewOptionalType(UFix64Type{}),
 		},
 		"nil": {
 			value:        NewOptional(nil),
 			string:       "nil",
+			goValue:      nil,
 			expectedType: NewOptionalType(NeverType{}),
 		},
 		"String": {
 			value:        String("Flow ridah!"),
 			string:       "\"Flow ridah!\"",
+			goValue:      "Flow ridah!",
 			expectedType: StringType{},
 		},
 		"Character": {
 			value:        Character("✌️"),
 			string:       "\"\\u{270c}\\u{fe0f}\"",
+			goValue:      "✌️",
 			expectedType: CharacterType{},
 		},
-		"Array": {
+		"StringArray": {
 			value: NewArray([]Value{
-				NewInt(10),
```

Comment:
Instead of changing the existing value, add an additional one

Reply:
The old test has an array that has both and int and a string, I can always add it back i just though it was very strange.